### ISSUE TITLE
Custom API for dynamically adding/modifying schemas

### ISSUE THREAD

**Issue by JPinkney** [redhat-developer/yaml-language-server#151]
This issue is to implement a custom API for dynamically adding and modifying JSON Schemas.

**Comment by JPinkney** [comment]
```
{
     "title": "CRD's",
     "properties": {
            "CRD Name": {
                  "type": "string",
                  "description": "The name of the CRD"
            }
     }
}
```

Suppose we have this JSON Schema that a client has added in via our new API.

The client might want to make a request that adds in another property to the schema to make the schema become:

```
{
     "title": "CRD's",
     "properties": {
            "CRD Name": {
                  "type": "string",
                  "description": "The name of the CRD"
            },
           "CRD Version": {
                  "type": "string",
                  "description": "The version of the CRD"
            }
     }
}
```

or perhaps the user wants to delete the "CRD Name" property and have:

```
{
     "title": "CRD's",
     "properties": {
           "CRD Version": {
                  "type": "string",
                  "description": "The version of the CRD"
            }
     }
}
```
instead.

And all features of the YAML Language server would update accordingly.

**Comment by JPinkney** [comment]
@gorkem Is the above what you had in mind for this new API?

**Comment by gorkem** [comment]
Yes this is the use case. In API terms We need the ability to pass a Schema object and also read the current one.

**Comment by JPinkney** [comment]
This is available as of the 0.7.0 release.

**Comment by mkatychev** [comment]
@JPinkney  is there an example of how this modification can be done?
